src/Storages/System/StorageSystemCertificates.cpp

```diff
@@ -0,0 +1,222 @@
+#include <Common/config.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Storages/System/StorageSystemCertificates.h>
+#include <regex>
+#include <filesystem>
+#include "Poco/File.h"
+#if USE_SSL
+    #include <openssl/x509v3.h>
+    #include "Poco/Net/SSLManager.h"
+    #include "Poco/Crypto/X509Certificate.h"
+#endif
+
+namespace DB
+{
+
+NamesAndTypesList StorageSystemCertificates::getNamesAndTypes()
+{
+    return
+    {
+        {"version",         std::make_shared<DataTypeNumber<Int32>>()},
+        {"serial_number",   std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"signature_algo",  std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"issuer",          std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"not_before",      std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"not_after",       std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"subject",         std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"pkey_algo",       std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"path",            std::make_shared<DataTypeString>()},
+        {"default",         std::make_shared<DataTypeNumber<UInt8>>()}
+    };
+}
+
+#if USE_SSL
+
+static std::unordered_set<std::string> parse_dir(const std::string & dir)
```

Comment:
If I understood correctly, we just want to split dir string by `':'` delimiter. Why don't just use `boost::split` instead of this code with loop and pointers?

Reply:
done

Reply:
Why do we need `std::unordered_set<std::string>` here? Is it because there may be repetitions?

Reply:
And I guess boost::split can handle `std::unordered_set<std::string>` as a first argument instead of vector

Reply:
yes, repetitions are possible and we don't want duplicates in the result